backend/backend.c

```diff
@@ -165,15 +168,32 @@ static struct wlr_backend *attempt_backend_by_name(struct wl_display *display,
 		if (strcmp(name, "libinput") == 0) {
 			return wlr_libinput_backend_create(display, *session);
 		} else {
-			return attempt_drm_backend(display, backend, *session);
+			return attempt_drm_backend(display, backend, *session, create_renderer_func);
 		}
 	}
 
 	wlr_log(L_ERROR, "unrecognized backend '%s'", name);
 	return NULL;
 }
 
-struct wlr_backend *wlr_backend_autocreate(struct wl_display *display) {
+struct wlr_renderer *wlr_backend_renderer_autocreate(struct wlr_egl *egl,
```

Comment:
This should be named `wlr_renderer_autocreate` and should be in `wlr/render/wlr_renderer.h`

Reply:
Should the implementation also be in `render/wlr_renderer.c`?

Reply:
I think so, yes.